compiler/vm/src/lir.rs

```diff
@@ -10,13 +11,15 @@ use candy_frontend::{
 use enumset::EnumSet;
 use extension_trait::extension_trait;
 use itertools::Itertools;
+use rustc_hash::FxHashSet;
 use std::fmt::{self, Display, Formatter};
 use strum::{EnumDiscriminants, IntoStaticStr};
 
 pub struct Lir {
     pub module: Module,
     pub constant_heap: Heap,
     pub instructions: Vec<Instruction>,
+    pub origins: Vec<FxHashSet<hir::Id>>,
```

Comment:
We should probably think about a more efficient storage later on, e.g., using run-length encoding